src/pages/CourtShow.js

```diff
@@ -1,43 +1,99 @@
-import player7 from "../assests/Player7.png"
-import player4 from "../assests/Player4.png"
+/* global google */
+import React, { useEffect, useState } from 'react';
+import { useParams, NavLink } from 'react-router-dom';
+import { Card, CardBody, CardTitle, CardSubtitle, Button } from 'reactstrap';
+import { useNavigate } from 'react-router-dom';
+import { GoogleMap, useJsApiLoader, Marker } from '@react-google-maps/api';
+import axios from 'axios';
+import player3 from '../assests/Player3.png';
+import player5 from '../assests/Player5.png';
+import player6 from '../assests/Player6.png';
+import player7 from '../assests/Player7.png';
+import player4 from '../assests/Player4.png';
 
 const CourtShow = ({ basketballcourts, deleteCourt, currentUser }) => {
+  const [formattedAddress, setFormattedAddress] = useState('');
+  const [center, setCenter] = useState({ lat: 0, lng: 0 });
+  const [courtDetails, setCourtDetails] = useState(null);
   let { id } = useParams();
   const currentCourt = basketballcourts?.find((court) => court.id === +id);
 
-  const players = [player3, player4, player5, player6, player7]
+  const players = [player3, player4, player5, player6, player7];
   const randomPlayers = players[Math.floor(Math.random() * players.length)];
 
-  const colors = ["#6BBAFF", "#FFA34A", "#FF5F5F", "#7052FF", "#19D79E"]
+  const colors = ['#6BBAFF', '#FFA34A', '#FF5F5F', '#7052FF', '#19D79E'];
   const randomColors = colors[Math.floor(Math.random() * colors.length)];
 
-  const navigate = useNavigate()
+  const navigate = useNavigate();
+
+  const { isLoaded } = useJsApiLoader({
+    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAPS_API_KEY,
+  });
+
+  const containerStyle = {
+    width: '400px',
+    height: '400px',
+  };
+
+  useEffect(() => {
+    if (currentCourt) {
+      geocodeAddress(currentCourt.address, currentCourt.city, currentCourt.state);
+    }
+  }, [currentCourt]);
+
+
+  const geocodeAddress = (street,  city, state ) => {
+    const address = `${street}, ${city}, ${state}`;
+    axios
+    .get(
+      `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${process.env.REACT_APP_GOOGLE_MAPS_API_KEY}`
+    )
+    .then((response) => {
+      if (response.data.results && response.data.results.length > 0) {
+        const rooftopResults = response.data.results.filter(result => result.types.includes('premise') || result.types.includes('street_address'));
+
+        if (rooftopResults.length > 0) {
+      const location = rooftopResults[0].geometry.location;
+      setCenter({ lat: location.lat, lng: location.lng });
+      setFormattedAddress(rooftopResults[0].formatted_address);
+     } else {
+      const approximateLocation = response.data.results.filter(result => result.geometry.location_type === 'APPROXIMATE')
+      const location = approximateLocation[0].geometry.location;
+      setCenter({ lat: location.lat, lng: location.lng });
+      setFormattedAddress(approximateLocation[0].formatted_address);
```

Comment:
Since we modified the values passed in the url, this section could be cleaned up and some of the code may not be needed anymore.